wgsl/index.bs

```diff
@@ -7272,13 +7270,337 @@ See [[#expressions]].
+
+When several edges have to be created we use `X -> {Y, Z}` as a short-hand for `X -> Y, X -> Z`.
+
+<table class='data'>
+  <caption>Uniformity rules for statements</caption>
+  <thead>
+    <tr><th>Statement<th>New nodes<th>Recursive analyses<th>Resulting control flow node<th>New edges
+  </thead>
+  <tr><td class="nowrap">{*s*}
+      <td>
+      <td class="nowrap">(*CF*, *s*) => *CF'*
+      <td>*CF'*
+      <td>
+  <tr><td class="nowrap">*s1* *s2*
+
+        Note: *s1* often ends in a semicolon.
+
+      <td>
+      <td class="nowrap">(*CF*, *s1*) => *CF1*<br>
+          (*CF1*, *s2*) => *CF2*
+      <td>*CF2*
+      <td>
+  <tr><td class="nowrap">if (*e*) *s1* else *s2*<br>, with behavior {Next}
+      <td>
+      <td rowspan=2 class="nowrap">(*CF*, *e*) => (*CF'*, *V*)<br>
+          (*V*, *s1*) => *CF1*<br>
+          (*V*, *s2*) => *CF2*
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">if (*e*) *s1* else *s2*<br>, with another behavior
+      <td>*CFend*
+      <td>*CFend*
+      <td class="nowrap">*CFend* -> {*CF1*, *CF2*}
+  <tr><td class="nowrap">loop {*s1* continuing {*s2*}}<br>, with behavior {Next}
+      <td rowspan=2>*CF'*
+      <td rowspan=2 class="nowrap">(*CF'*, *s1*) => *CF1*<br>
+          (*CF1*, *s2*) => *CF2*
+      <td>*CF*
+      <td rowspan=2 class="nowrap">*CF'* -> {*CF2*, *CF*}
+  <tr><td class="nowrap">loop {*s1* continuing {*s2*}}<br>, with another behavior
+      <td>*CF'*
+  <tr><td class="nowrap">loop {*s1*}, with behavior {Next}
+      <td rowspan=2>*CF'*
+      <td rowspan=2 class="nowrap">(*CF'*, *s1*) => *CF1*
+      <td>*CF*
+      <td rowspan=2 class="nowrap">*CF'* -> {*CF1*, *CF*}
+  <tr><td class="nowrap">loop {*s1*}, with another behavior
+      <td>*CF'*
+  <tr><td class="nowrap">switch(*e*) case _: *s_1* .. case _: *s_n*<br>, with behavior {Next}
+      <td>
+      <td rowspan=2 class="nowrap">(*CF*, *e*) => (*CF'*, *V*)<br>
+        (*V*, *s_1*) => *CF_1*<br>
+        ...<br>
+        if *s_(n-1)* may fallthrough, (*CF_(n-1)*, *s_n*) => *CF_n*<br>
+        else (*V*, *s_n*) => *CF_n*
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">switch(*e*) case _: *s_1* .. case _: *s_n*<br>, with another behavior
+      <td>*CFend*
+      <td>*CFend*
+      <td class="nowrap">*CFend* -> {*CF_1*, ..., *CF_n*}
+  <tr><td class="nowrap">var x: T;
+      <td rowspan=5>
+      <td rowspan=5>
+      <td rowspan=5>*CF*
+      <td rowspan=5>
+  <tr><td class="nowrap">break;
+  <tr><td class="nowrap">continue;
+  <tr><td class="nowrap">fallthrough;
+  <tr><td class="nowrap">discard;
+  <tr><td class="nowrap">return;
+      <td>
+      <td>
+      <td>*CF*
+      <td>*CF_return* -> *CF*
+  <tr><td class="nowrap">return *e*;
+      <td>
+      <td class="nowrap">(*CF*, *e*) => (*CF'*, *V*)
+      <td>*CF'*
+      <td>*CF_return* -> *CF'*<br>
+          *Value_return* -> *V*
+  <tr><td class="nowrap">*e1* = *e2*;
+      <td>
+      <td class="nowrap">LValue: (*CF*, *e1*) => (*CF1*, *L1*)<br>
```

Comment:
This has the LHS being evaluated before the RHS.
Current spec says the RHS is evaluated before the LHS.
We can fix this later.

Reply:
Ah, good catch, I kept expecting left-to-right evaluation. I'll fix it.